### PR TITLE
Index Results Template

### PR DESCRIPTION
**Pull request by isahwahni**
This PR contains the following:
- Adding template Report for Execution Results Index.
- updating the code to reflect the changes.
- updating updateEndTime function to reflect the new format of the index template.
- Code cleanup for Static HTML writing.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ibatta/SelFW/pull/7%23discussion_r151861971%22%2C%20%22https%3A//github.com/ibatta/SelFW/pull/7%23discussion_r151943750%22%2C%20%22https%3A//github.com/ibatta/SelFW/pull/7%23issuecomment-345702762%22%2C%20%22https%3A//github.com/ibatta/SelFW/pull/7%23issuecomment-345704226%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ibatta/SelFW/pull/7%23issuecomment-345702762%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40ibatta%20%20The%20updateEndTime%20is%20reverted%20to%20its%20original%20implementation%20in%20order%20to%20handle%20runners%20%22%2C%20%22created_at%22%3A%20%222017-11-20T13%3A57%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/33652900%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/isahwahni%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222017-11-20T14%3A03%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/18301319%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ibatta%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2095f35694c46fafa90ca164252a49aee7496cfb12%20src/com/generic/util/ReportResultsWriter.java%2075%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ibatta/SelFW/pull/7%23discussion_r151861971%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40isahwahni%20%20replace%20index%20of%20with%20contains%22%2C%20%22created_at%22%3A%20%222017-11-19T11%3A36%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/18301319%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ibatta%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222017-11-20T09%3A50%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/33652900%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/isahwahni%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/com/generic/util/ReportResultsWriter.java%3AL185-223%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 95f35694c46fafa90ca164252a49aee7496cfb12 src/com/generic/util/ReportResultsWriter.java 75'></a> &nbsp;<b><a href='https://github.com/ibatta/SelFW/pull/7#discussion_r151861971'>File: src/com/generic/util/ReportResultsWriter.java:L185-223</a></b>
- <a href='https://github.com/ibatta'></a> @isahwahni  replace index of with contains
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ibatta/SelFW/pull/7#issuecomment-345702762'>General Comment</a></b>
- <a href='https://github.com/isahwahni'></a> @ibatta  The updateEndTime is reverted to its original implementation in order to handle runners

<a href='https://www.codereviewhub.com/ibatta/SelFW/pull/7?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ibatta/SelFW/pull/7?approve=1'></a>&nbsp;<a href='https://github.com/ibatta/SelFW/pull/7'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ibatta** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ibatta/SelFW/pull/7'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/com/generic/setup/EnvironmentFiles.java [hunk 1]
```diff
@@ -5,6 +5,18 @@
 public class EnvironmentFiles {
 	//config.properties file path,getter and setter
 	public static String configFilePath = System.getProperty("user.dir")+"//src//com//generic//config//config.properties";
+	//DataSheet.xlsx file path,getter and setter
+	public static String dataSheetPath = System.getProperty("user.dir")+"//src//com//generic//config//DataSheet.xlsx";
+	//log file path,getter and setter
+	public static String logFilePath = null;
+	//log file name,getter and setter
+	public static String logFileName = "Application.log";
```
Comment (ibatta) [inline src/com/generic/setup/EnvironmentFiles.java]: @isahwahni  we need to make this dynamic naming

File: src/com/generic/setup/EnvironmentFiles.java [hunk 2]
```diff
@@ -68,15 +68,26 @@ public static void setReportsFolderPath(String reportsFolderPath) {
...
 	}
 
 	public static void setFailedTestCaseReportTemplate(String failedTestCaseReportTemplate) {
 		EnvironmentFiles.failedTestCaseReportTemplate = failedTestCaseReportTemplate;
 	}
+
+	/**
+	 * @return the testCasesIndexReportTemplate
+	 */
+	public static String getTestCasesIndexReportTemplate() {
```
Comment (ibatta) [inline src/com/generic/setup/EnvironmentFiles.java]: @isahwahni  refactor this

File: src/com/generic/util/ReportResultsWriter.java [hunk 3]
```diff
@@ -147,7 +147,7 @@ public static void generateFailedTestCasePage(final String testCaseName, FileWri
 			// copying template into target
 			String str;
 		    while ((str = br.readLine()) != null) {
-		    	SelTestCase.logs.debug("Writing the tag : \n" + str);
+//		    	SelTestCase.logs.debug("Writing the tag : \n" + str);
```
Comment (ibatta) [inline src/com/generic/util/ReportResultsWriter.java]: @isahwahni  remove it

File: src/com/generic/util/ReportResultsWriter.java [hunk 4]
```diff
@@ -185,57 +185,38 @@ public static void writeTestCasesHeader(final String suiteName, BufferedWriter o
...
+			BufferedReader br = new BufferedReader(new FileReader(EnvironmentFiles.getTestCasesIndexReportTemplate()));
 			
+			// copying template into target
+			String str;
+		    while ((str = br.readLine()) != null) {
+		    	SelTestCase.logs.debug("Writing the tag : \n" + str);
+		    	if (str.indexOf("~RunDate~") != -1) {
+		    		str=str.replaceAll("~RunDate~", rUNDATE);
+                	
+                } if (str.indexOf("~Env~") != -1) {
```
Comment (ibatta) [inline src/com/generic/util/ReportResultsWriter.java]: @isahwahni  replace index of with contains

File: src/com/generic/util/ReportUtil.java [hunk 5]
```diff
@@ -171,7 +171,7 @@ public static void startSuite(final String suiteName) {
             fstream = new FileWriter(indexResultFilename, true);
             out = new BufferedWriter(fstream);
 
-            ReportResultsWriter.writeTestCasesHeader(suiteName, out);
+//            ReportResultsWriter.writeTestCasesHeader(suiteName, out);
```
Comment (ibatta) [inline src/com/generic/util/ReportUtil.java]: @isahwahni  clean it up

File: src/com/generic/util/ReportUtil.java [hunk 6]
```diff
@@ -315,9 +315,9 @@ public static void updateEndTime(final String endTime, String testStat) {
 
             while ((strLine = br.readLine()) != null) {
 
-                if (strLine.indexOf("end_time") != -1) {
+                if (strLine.indexOf("~EndTime~") != -1) {
```
Comment (ibatta) [inline src/com/generic/util/ReportUtil.java]: @isahwahni  replace it with contains

File: src/com/generic/util/ReportResultsWriter.java (hunk 4 above)
Comment (isahwahni) [inline src/com/generic/util/ReportResultsWriter.java]: :+1:

**Comment by isahwahni** [comment]
@ibatta  The updateEndTime is reverted to its original implementation in order to handle runners

**Comment by ibatta** [comment]
:+1:

**Pull request merged**